### ISSUE TITLE
Better non-Director shortcut for isDev()

### ISSUE THREAD

**Issue by sminnee** [silverstripe/silverstripe-framework#9685]
This flows out of my work on #9679 

Currently, `Director::isDev()` is used heavily by code that shouldn't be coupled to Control.

The alternative is to use this, which is a mouthful:
```php
Injector::inst()->get(Kernel::class)->getEnvironment() === 'dev'
```

To decouple while retaining a decent dev experience, I'd like to have a shorthand. The Environment class has a number of static helper methods and so I think  `Environment::isDev()` and other companions would make sense. It would call the above.

It might also make sense to introduce `Environment::getKernel()` for completeness.

**Comment by emteknetnz** [comment]
Seems fine

So presumably to retain backwards compatibility we would:

```php
**Director.php**

public static function isDev() {
    return Environment::isDev();
}
```
?

We'd also want Environment::isTest() and Environment::isLive() ?

**Comment by sminnee** [comment]
Yes, and yes, and get_environment_type. :)

There's also currently code that calls `Director::get_session_environment_type` to allow isDev = 1. The simplest fix would be to put `if (class_exists(Director::class))` as a wrapper around that, which is probably what I'd leave this as for initially. In time it would be handy if this facility could be loaded in by the Director module, but that can come later. It's arguably a feature that should be dropped as it's a bit of a security hazard.

**Comment by ScopeyNZ** [comment]
+1 to belonging to `Environment`. Getting the kernel from the environment is questionable if we're being strict, as the same environment should be able to run different kernels, but I'm not really too fussed with that.

**Comment by sminnee** [comment]
Yeah Enviroment is just a bucket of static method, and this will just be syntactic sugar for accessing the kernel through injector, so I think it's okay.

It might be a bit heavy-handed to throw deprecation notices for Director::isDev though?

**Comment by robbieaverill** [comment]
IMO we shouldn't be afraid to deprecate things. Maybe we need to encourage devs to disable deprecation warnings (from SilverStripe) if it's a problem for devs to work with. Otherwise we could stop using our deprecation API and just use the phpdoc annotations instead.

**Comment by sminnee** [comment]
Yeah; I think this might deserve a separate ticket to look at status quo and clarify recommended approach.

I'm just aware that Director::isDev isn't exactly an obscure edge-case API and upgrades should be painless
